### PR TITLE
Use binary mode to open files in base storage

### PR DESCRIPTION
**Pull request by claudep**
This is in line with commit 2f889c19e2.
Also marked with b/bytes some more strings.

### CODE REVIEW

File: translate/storage/base.py [hunk 1]
```diff
@@ -795,13 +792,10 @@ def save(self):
         """Save to the file that data was originally read from, if
         available."""
         fileobj = getattr(self, "fileobj", None)
-        mode = 'w'
-        if self._binary:
-            mode = 'wb'
+        mode = 'wb'
```
Comment (unho) [inline translate/storage/base.py]: It is preferable to avoid creating variables that are not strictly necessary.

Comment (unho) [inline translate/storage/base.py]: Also having the mode string in the `open` call (or wherever the mode is used) makes it easier to understand the code.

**Comment by unho** [comment]
Next time please try to keep these changes in separate commits.

**Comment by claudep** [comment]
Rebased and commit separated.

**Comment by unho** [comment]
Merged. Thanks!

**Pull request merged**